packages/video_player/video_player_avfoundation/example/ios/RunnerTests/VideoPlayerTests.m

```diff
@@ -396,6 +398,111 @@ - (void)testSeekToleranceWhenSeekingToEnd {
   return initializationEvent;
 }
 
+// Checks whether [AVPlayer rate] KVO observations are correctly detached.
+// - https://github.com/flutter/flutter/issues/124937
+//
+// Failing to de-register results in a crash in [AVPlayer willChangeValueForKey:].
+- (void)testDoesNotCrashOnRateObservationAfterDisposal {
+  NSObject<FlutterPluginRegistry> *registry =
+      (NSObject<FlutterPluginRegistry> *)[[UIApplication sharedApplication] delegate];
+  NSObject<FlutterPluginRegistrar> *registrar =
+      [registry registrarForPlugin:@"testDoesNotCrashOnRateObservationAfterDisposal"];
+
+  AVPlayer *avPlayer = nil;
+  __weak FLTVideoPlayer *player = nil;
+
+  // Autoreleasepool is needed to simulate conditions of FLTVideoPlayer deallocation.
+  @autoreleasepool {
+    FLTVideoPlayerPlugin *videoPlayerPlugin =
+        (FLTVideoPlayerPlugin *)[[FLTVideoPlayerPlugin alloc] initWithRegistrar:registrar];
+
+    FlutterError *error;
+    [videoPlayerPlugin initialize:&error];
+    XCTAssertNil(error);
+
+    FLTCreateMessage *create = [FLTCreateMessage
+        makeWithAsset:nil
+                  uri:@"https://flutter.github.io/assets-for-api-docs/assets/videos/bee.mp4"
+          packageName:nil
+           formatHint:nil
+          httpHeaders:@{}];
+    FLTTextureMessage *textureMessage = [videoPlayerPlugin create:create error:&error];
+    XCTAssertNil(error);
+    XCTAssertNotNil(textureMessage);
+
+    player = videoPlayerPlugin.playersByTextureId[textureMessage.textureId];
+    XCTAssertNotNil(player);
+    avPlayer = player.player;
+
+    [videoPlayerPlugin dispose:textureMessage error:&error];
+    XCTAssertNil(error);
+  }
+
+  // [FLTVideoPlayerPlugin dispose:error:] selector is dispatching the [FLTVideoPlayer dispose] call
+  // with a 1-second delay keeping a strong reference to the player. The polling ensures the player
```

Comment:
what does this 1 sec delay come from?

Reply:
The `[FLTVideoPlayerPlugin dispose:error:]` has this piece of code:

```
dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1 * NSEC_PER_SEC)),
               dispatch_get_main_queue(), ^{
                 if (!player.disposed) {
                   [player dispose];
                 }
               });
```